arbos/util/transfer.go

```diff
@@ -33,14 +34,14 @@ func TransferBalance(
 		if arbmath.BigLessThan(balance.ToBig(), amount) {
 			return fmt.Errorf("%w: addr %v have %v want %v", vm.ErrInsufficientBalance, *from, balance, amount)
 		}
-		evm.StateDB.SubBalance(*from, uint256.MustFromBig(amount))
+		evm.StateDB.SubBalance(*from, uint256.MustFromBig(amount), tracing.BalanceChangeTransfer)
 		if evm.Context.ArbOSVersion >= 30 {
 			// ensure the from account is "touched" for EIP-161
-			evm.StateDB.AddBalance(*from, &uint256.Int{})
+			evm.StateDB.AddBalance(*from, &uint256.Int{}, tracing.BalanceChangeTransfer)
 		}
 	}
 	if to != nil {
-		evm.StateDB.AddBalance(*to, uint256.MustFromBig(amount))
+		evm.StateDB.AddBalance(*to, uint256.MustFromBig(amount), tracing.BalanceChangeTransfer)
 	}
 	if tracer := evm.Config.Tracer; tracer != nil {
 		if evm.Depth() != 0 && scenario != TracingDuringEVM {
```

Comment:
Previously, before 1.14.0, geth used to first do value transfer (from sender to receiver) and then call tracing as seen [here](https://github.com/OffchainLabs/go-ethereum/blob/48de2030c7a6fa8689bc0a0212ebca2a0c73e3ad/core/vm/evm.go#L222-L239),  but now it first calls tracing and then does value transfer as seen [here](https://github.com/OffchainLabs/go-ethereum/blob/merge-1.14.0/core/vm/evm.go#L199-L223).

I think we should match what geth does, this is specifically important in cases when tracing with prestate tracers.

Reply:
Done, thanks for catching this!